src/CppUTest/TestFailure.cpp

```diff
@@ -318,6 +318,28 @@ BinaryEqualFailure::BinaryEqualFailure(UtestShell* test, const char* fileName, i
 	}
 }
 
+AnyIntsEqualFailure::AnyIntsEqualFailure(UtestShell* test, const char* fileName, int lineNumber, const unsigned char* expected,
+                                       const unsigned char* actual, size_t size, const SimpleString& text)
+: TestFailure(test, fileName, lineNumber)
+{
+    if (size <= sizeof(long))
+    {
+        long lexp = 0, lact = 0;
+        for (size_t i = 0; i < size; i++)
+            ((unsigned char*)&lexp)[i] = ((const unsigned char *)expected)[i];
```

Comment:
Can you add this loop in a helper function (inside the class, private) that describes what it does and call it twice?